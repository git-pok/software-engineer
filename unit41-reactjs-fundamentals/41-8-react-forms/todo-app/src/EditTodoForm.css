.EditTodoForm-form {
  width: 50vw;
  height: 150px;
  padding-top: 30px;
  padding-left: 30px;
  margin-bottom: 30px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  background-color: rgb(22, 21, 40);
}

.EditTodoForm-input {
  margin-bottom: 20px;
  width: 290px;
  height: 25px;
  border-radius: 10px;
  font-size: 15px;
}

.EditTodoForm-label {
  font-size: 25px;
  color: rgb(237, 220, 169);
  margin-bottom: 5px;
}

.EditTodoForm-submit {
  width: 110px;
  height: 25px;
  border-radius: 10px;
  border: 2px solid rgb(255, 255, 255);
  background-color: rgb(232, 169, 237);
}