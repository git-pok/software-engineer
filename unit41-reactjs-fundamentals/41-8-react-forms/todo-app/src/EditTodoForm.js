import { useState } from 'react';
import './EditTodoForm.css';

/**
 * EditTodoForm - this component should render a form
 * with one text input for the task to be edited. When
 * this form is submitted, a Todo component
 * should be edited.
 */


const EditTodoForm = ({ editTodo, id }) => {

  const INITIAL_STATE = {
    todo: ""
  };

  const [ formData, setFormData ] = useState(INITIAL_STATE);

  const handleChange = e => {
    const evtTarget = e.target; 
    const { name, value } = evtTarget;

    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  const handleSubmit = e => {
    e.preventDefault();
    const edit = formData.todo;
    editTodo({ id, edit });
    setFormData(INITIAL_STATE);
  }

  return (
    <form onSubmit={handleSubmit} className="EditTodoForm-form">
      <label
        className="EditTodoForm-label"
        htmlFor="todo">Edit Todo</label>
      <input
        className="EditTodoForm-input"
        id="todo"
        type="text"
        name="todo"
        placeholder="Edit Todo!!!"
        value={formData.todo}
        onChange={handleChange} />

      <button className="EditTodoForm-submit">SUBMIT</button>
    </form>
  );
}

export default EditTodoForm;