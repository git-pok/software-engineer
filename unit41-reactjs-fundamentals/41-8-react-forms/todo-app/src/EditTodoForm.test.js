import { render } from '@testing-library/react';
import EditTodoForm from './EditTodoForm.js'; 


test('<EditTodoForm /> renders without crashing', () => {
  render(<EditTodoForm />);
});


test('fragment matches snapshot', () => {
  const { asFragment } = render(<EditTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});