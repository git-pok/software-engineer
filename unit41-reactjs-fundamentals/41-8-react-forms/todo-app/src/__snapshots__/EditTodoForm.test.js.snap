// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditTodoForm-form"
  >
    <label
      class="EditTodoForm-label"
      for="todo"
    >
      Edit Todo
    </label>
    <input
      class="EditTodoForm-input"
      id="todo"
      name="todo"
      placeholder="Edit Todo!!!"
      type="text"
      value=""
    />
    <button
      class="EditTodoForm-submit"
    >
      SUBMIT
    </button>
  </form>
</DocumentFragment>
`;
